pkg/openfeature/provider.go

```diff
@@ -60,6 +65,62 @@ type StateHandler interface {
 	Status() State
 }
 
+// NoopStateHandler is a noop StateHandler implementation
+// Status always set to ReadyState to comply with specification
+type NoopStateHandler struct {
+}
+
+func (s *NoopStateHandler) Init(e EvaluationContext) {
+	// NOOP
+}
+
+func (s *NoopStateHandler) Shutdown() {
+	// NOOP
+}
+
+func (s *NoopStateHandler) Status() State {
+	return ReadyState
+}
+
+// Eventing
+
+// EventHandler is the eventing contract enforced for FeatureProvider
+type EventHandler interface {
+	EventChannel() <-chan Event
+}
+
+// EventType emitted by a provider implementation
+type EventType string
+
+// ProviderEventDetails is the event payload emitted by FeatureProvider
+type ProviderEventDetails struct {
+	Message       string
+	FlagChanges   []string
+	EventMetadata map[string]interface{}
+}
+
+// Event is an event emitted by a FeatureProvider.
+type Event struct {
+	ProviderName string
+	EventType
+	ProviderEventDetails
+}
+
+type EventDetails struct {
+	client string
+	ProviderEventDetails
+}
+
+type EventCallBack *func(details EventDetails)
```

Comment:
```suggestion
type EventCallback *func(details EventDetails)
```
I'd consider `handler` but callback is fine too. I would do whatever _feels_ more right for Go users.

If you do use callback though, note that it's one word (https://en.wikipedia.org/wiki/Callback_(computer_programming)).

Reply:
Thanks for lang improvement. Updated to `EventCallback `